Dockerfile

```diff
@@ -1,19 +1,44 @@
 # Dockerfile to build and serve datagateway-api
 
-FROM python:3.6-slim-bullseye
+# Build stage
+FROM python:3-alpine as builder
 
-WORKDIR /datagateway-api
+WORKDIR /datagateway-api-build
 
-COPY poetry.lock pyproject.toml ./
+COPY . .
 
-RUN python -m pip install --upgrade pip \
-  && pip install 'poetry==1.1.13' \
-  && poetry run pip uninstall -y setuptools \
-  && poetry run pip install 'setuptools<58.0.0' \
-  && poetry run pip install 'gunicorn==20.1.0' \
-  && poetry install --no-dev
+RUN python3 -m pip install poetry \
+ && poetry build
 
-COPY datagateway_api ./datagateway_api
+
+# Install & run stage
+FROM python:3-alpine
```

Comment:
Best to pin the version of Python to prevent unexpected breakages in future.

Reply:
Resolved in https://github.com/ral-facilities/datagateway-api/commit/dfb54bac1808117c93dd7bcab4b3c4f8a0919786